huaweicloud/services/acceptance/ims/resource_huaweicloud_images_image_share_test.go

```diff
@@ -0,0 +1,129 @@
+	if err != nil {
+		return nil, fmt.Errorf("error creating IMS Client: %s", err)
+	}
+
+	getImagePath := getImageClient.Endpoint + getImageHttpUrl
+	getImagePath = strings.ReplaceAll(getImagePath, "{project_id}", getImageClient.ProjectID)
+
+	imageId := state.Primary.Attributes["source_image_id"]
+	getImageQueryParams := buildGetImageQueryParams(imageId)
+	getImagePath += getImageQueryParams
+
+	getImageOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getImageResp, err := getImageClient.Request("GET", getImagePath, &getImageOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving IMS image: %s", err)
+	}
+
+	getImageRespBody, err := utils.FlattenResponse(getImageResp)
+	if err != nil {
+		return nil, err
+	}
+
+	images := utils.PathSearch("images", getImageRespBody, nil)
+	if images == nil || len(images.([]interface{})) == 0 {
+		return nil, fmt.Errorf("error get IMS share image")
+	}
+
+	return make(map[string]interface{}), nil
+}
+
+func buildGetImageQueryParams(id string) string {
+	res := ""
+	res = fmt.Sprintf("%s?id=%v", res, id)
+	res = fmt.Sprintf("%s&__imagetype=%v", res, "shared")
+	return res
+}
+
+func TestAccImsImageShare_basic(t *testing.T) {
+	var obj interface{}
+
+	imageName := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_images_image_share.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getImsImageShareResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheckReplication(t)
+			acceptance.TestAccPreCheckEpsID(t)
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testImsImageShare_basic(imageName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "source_image_id",
+						"huaweicloud_images_image.test", "id"),
+				),
+			},
+			{
+				Config: testImsImageShare_update(imageName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "source_image_id",
+						"huaweicloud_images_image.test", "id"),
+				),
+			},
+		},
+	})
+}
+
+func testImsImageShare_basic(imageName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_images_image_share" "test" {
+ source_image_id    = huaweicloud_images_image.test.id
+ target_project_ids = ["%[2]s"]
+}
+`, testAccImsImage_basic(imageName), acceptance.HW_DEST_PROJECT_ID)
+}
+
+func testImsImageShare_update(imageName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_images_image_share" "test" {
+ source_image_id    = huaweicloud_images_image.test.id
+ target_project_ids = ["%[2]s"]
+}
+`, testAccImsImage_basic(imageName), acceptance.HW_ENTERPRISE_PROJECT_ID_TEST)
```

Comment:
HW_ENTERPRISE_PROJECT_ID_TEST is not the project id

Reply:
done